pubsub/natsstreaming/natsstreaming.go

```diff
@@ -290,6 +317,7 @@ func genRandomString(n int) string {
 }
 
 func (n *natsStreamingPubSub) Close() error {
+	n.cancel()
```

Comment:
similar to this, close background context for Kafka pubusb?

Reply:
In the Kafka pubsub, `closeSubscripionResources` handles calling cancel because it creates a new context/cancel in each call to `Subscribe`. It does feel like maybe the context should live for the duration of the component and not for the subscription. The feels large enough for its own issue and PR but I :100: agree.